compiler/plugins/target/CUDA/CUDATarget.cpp

```diff
@@ -66,37 +66,33 @@ struct CUDAOptions {
     binder.opt<bool>("iree-hal-cuda-dump-ptx", dumpPtx, llvm::cl::cat(category),
                      llvm::cl::desc("Dump ptx to the debug stream."));
 
-    binder.opt<std::string>("iree-hal-cuda-llvm-target-arch", clTargetChip,
+    binder.opt<std::string>("iree-cuda-target", clTarget,
```

Comment:
I'm not sure about the namespace here.

Some options to consider:

* `iree-hal-cuda-` namespace under `hal` since these relate to the hal part of the pipeline
* `iree-plugin-cuda-` namespace under `plugin` since this code is in a plugin
* `iree-cuda-` floating near the top of the namespace hierarchy

Reply:
I treat these namespaces as mostly a way to avoid command-line option collison; not to indicate strong nested logical/directory structure. Reason being that from a developer's perspective, it does not matter much to know that, oh, cuda/hip/vulkan is a plugin under hal, which is more implementation details. 

Also given these APIs are top tier we are supporting, it's okay to directly use `-iree-<api>` as a prefix imo--it's shorter and easier to memorize. The `iree-hal` category can leave for things common to all hal impls (if we ever have).

we sort of doing this structure on codegen side, where `iree-codegen` prefix is for stuff common to all codegen backends, while `iree-spirv`, `iree-llvmgpu`, etc. for backend specific ones. wdyt?

Reply:
Using `--iree-<api>-*` instead of `--iree-hal-<api>-*` sounds good to me. @benvanik might have an opinion too.